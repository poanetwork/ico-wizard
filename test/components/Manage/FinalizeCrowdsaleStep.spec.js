import React from 'react'
import { StaticRouter } from 'react-router'
import { FinalizeCrowdsaleStep } from '../../../src/components/Manage/FinalizeCrowdsaleStep'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount } from 'enzyme'

configure({ adapter: new Adapter() })

describe('FinalizeCrowdsaleStep', () => {
  it('should render the component with active button', () => {
    const finalizeCrowdsaleStateParams = {
      disabled: false,
      handleClick: jest.fn()
    }

    expect(
      renderer
        .create(
          <StaticRouter location="testLocation" context={{}}>
            <FinalizeCrowdsaleStep {...finalizeCrowdsaleStateParams} />
          </StaticRouter>
        )
        .toJSON()
    ).toMatchSnapshot()
  })

  it('should render the component with disabled button', () => {
    const finalizeCrowdsaleStateParams = {
      disabled: true,
      handleClick: jest.fn()
    }

    expect(
      renderer
        .create(
          <StaticRouter location="testLocation" context={{}}>
            <FinalizeCrowdsaleStep {...finalizeCrowdsaleStateParams} />
          </StaticRouter>
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
