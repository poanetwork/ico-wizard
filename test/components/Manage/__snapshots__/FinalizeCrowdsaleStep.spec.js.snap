// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalizeCrowdsaleStep should render the component with active button 1`] = `
<div
  className="mng-FinalizeCrowdsaleStep"
>
  <div
    className="mng-FinalizeCrowdsaleStep_Description"
  >
    After finalization, it’s not possible to update tiers or buy tokens. All tokens will be movable and reserved tokens will be issued.
  </div>
  <button
    className="mng-FinalizeCrowdsaleStep_Button"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Finalize Crowdsale
  </button>
</div>
`;

exports[`FinalizeCrowdsaleStep should render the component with disabled button 1`] = `
<div
  className="mng-FinalizeCrowdsaleStep"
>
  <div
    className="mng-FinalizeCrowdsaleStep_Description"
  >
    After finalization, it’s not possible to update tiers or buy tokens. All tokens will be movable and reserved tokens will be issued.
  </div>
  <button
    className="mng-FinalizeCrowdsaleStep_Button"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Finalize Crowdsale
  </button>
</div>
`;
